libs/math/include/math/utilities/ReadCSV.hpp

```diff
@@ -94,7 +94,16 @@ TensorType ReadCSV(std::string const &filename, math::SizeType const cols_to_ski
     }
     while (std::getline(ss, field_value, delimiter))
     {
-      weights(col, row) = fetch::math::Type<DataType>(field_value);
+      if (!unsafe_parsing)
+      {
```

Comment:
I would swap these two blocks and eliminate "!" at the (!unsafe_parsing), because this "!" is very easy to miss while fast-reading the code.